pkg/odo/cli/url/create.go

```diff
@@ -0,0 +1,121 @@
+package url
+
+import (
+	"fmt"
+
+	"github.com/redhat-developer/odo/pkg/log"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/url"
+
+	odoutil "github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/util"
+	"github.com/spf13/cobra"
+
+	ktemplates "k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+)
+
+const createRecommendedCommandName = "create"
+
+var (
+	urlCreateShortDesc = `Create a URL for a component`
+	urlCreateLongDesc  = ktemplates.LongDesc(`Create a URL for a component.
+
+	The created URL can be used to access the specified component from outside the OpenShift cluster.
+	`)
+	urlCreateExample = ktemplates.Examples(`  # Create a URL for the current component with a specific port
+	%[1]s --port 8080
+  
+	# Create a URL with a specific name and port
+	%[1]s example --port 8080
+  
+	# Create a URL with a specific name by automatic detection of port (only for components which expose only one service port) 
+	%[1]s example
+  
+	# Create a URL with a specific name and port for component frontend
+	%[1]s example --port 8080 --component frontend
+	  `)
+)
+
+// URLCreateOptions encapsulates the options for the odo url create command
+type URLCreateOptions struct {
+	urlName       string
+	urlPort       int
+	urlOpenFlag   bool
+	componentPort int
+	*genericclioptions.Context
+}
+
+// NewURLCreateOptions creates a new UrlCreateOptions instance
+func NewURLCreateOptions() *URLCreateOptions {
+	return &URLCreateOptions{}
+}
+
+// Complete completes UrlCreateOptions after they've been Created
+func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.Context = genericclioptions.NewContext(cmd)
+	o.componentPort, err = url.GetValidPortNumber(o.Client, o.urlPort, o.Component(), o.Application)
+	if err != nil {
+		return err
+	}
+	switch len(args) {
+	case 0:
+		o.urlName = url.GetURLName(o.Component(), o.componentPort)
+	case 1:
+		o.urlName = args[0]
+	default:
+		return fmt.Errorf("unable to get component")
+	}
+	return
+}
+
+// Validate validates the UrlCreateOptions based on completed values
+func (o *URLCreateOptions) Validate() (err error) {
+	exists, err := url.Exists(o.Client, o.urlName, "", o.Application)
```

Comment:
I understand that this was like this before this refactoring but why aren't we passing the component name here?